datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -3589,6 +3635,18 @@ mod tests {
         let expr = not_ilike(col("c1"), "%");
         assert_eq!(simplify(expr), lit(false));
 
+        let expr = like(col("c1"), "a%");
```

Comment:
add test cases

- with `%` pattern 
- with `_` pattern (can be simplified to a length check, but that can be follow up)

- with `%%` and `%%%` patterns (can be simplified just like single `%`)
- with `__` and `___` patterns (can be simplified to a length check, but that can be follow up)
- with patterns combining % and _, eg `a_b%`, `a%b_` (these cannot simplified) and  `_%`,  `%_`  (these can be simplified to a length check, but that can be follow up)